integration-tests/manifests/annotations/validate_annotations_test.go

```diff
@@ -0,0 +1,1003 @@
+		t.Error("Incorrect Annotations")
+	}
+}
+
+func applyYAMLWithKubectl(filename, namespace string) error {
+	cmd := exec.Command("kubectl", "apply", "-f", filename, "-n", namespace)
+	return cmd.Run()
+}
+
+// Usage in your createNamespace function
+func createNamespaceAndApplyResources(t *testing.T, clientset *kubernetes.Clientset, name string, resourceFiles []string) error {
+	// Create Namespace
+	err := createNamespace(clientset, name)
+	if err != nil {
+		return err
+	}
+
+	time.Sleep(25 * time.Second)
+	// Apply each YAML file
+	for _, file := range resourceFiles {
+		err = applyYAMLWithKubectl(filepath.Join("..", file), name)
+		if err != nil {
+			t.Error("Could not apply resources")
+			return err
+		}
+	}
+	time.Sleep(25 * time.Second)
+
+	return nil
+}
+
+func deleteYAMLWithKubectl(filename, namespace string) error {
+	cmd := exec.Command("kubectl", "delete", "-f", filename, "-n", namespace)
+	return cmd.Run()
+}
+
+func deleteNamespaceAndResources(clientset *kubernetes.Clientset, name string, resourceFiles []string) error {
+	// Delete each YAML file
+	for _, file := range resourceFiles {
+		err := deleteYAMLWithKubectl(filepath.Join("..", file), name)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Delete Namespace
+	err := deleteNamespace(clientset, name)
+	time.Sleep(25 * time.Second)
+	return err
+}
+func createNamespace(clientset *kubernetes.Clientset, name string) error {
+	// Check if the namespace already exists
+	_, err := clientset.CoreV1().Namespaces().Get(context.Background(), name, metav1.GetOptions{})
+	if err == nil {
+		// Namespace exists - you can choose to return nil or delete and recreate it
+		return nil // Use this line if you want to just use the existing namespace
+		// Uncomment below lines if you want to delete and recreate the namespace
+		// err = deleteNamespace(clientset, name)
+		// if err != nil {
+		// return err
+		// }
+	} else if !errors.IsNotFound(err) {
+		// Other error while getting namespace
+		return err
+	}
+
+	// Create the namespace if it doesn't exist
+	namespace := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: name}}
+	_, err = clientset.CoreV1().Namespaces().Create(context.Background(), namespace, metav1.CreateOptions{})
+	return err
+}
+
+func deleteNamespace(clientset *kubernetes.Clientset, name string) error {
+	return clientset.CoreV1().Namespaces().Delete(context.Background(), name, metav1.DeleteOptions{})
+}
+
+func checkNameSpaceAnnotations(ns *v1.Namespace, expectedAnnotations []string) bool {
+	for _, annotation := range expectedAnnotations {
+		if ns.ObjectMeta.Annotations[annotation] != "true" {
+			return false
+		}
+	}
+	fmt.Println("Namespace annotations are correct!")
+	return true
+}
+func updateOperator(t *testing.T, clientSet *kubernetes.Clientset, deployment *appsV1.Deployment) bool {
+	opMutex.Lock() // Lock the mutex at the beginning
+	defer opMutex.Unlock()
+	var err error
+	args := deployment.Spec.Template.Spec.Containers[0].Args
```

Comment:
What's the reason that we can't just update operator using the `deployment` variable as it is if it's already updated with the new arg?